Products/validation/i18n.py

```diff
@@ -21,6 +21,15 @@ def recursiveTranslate(message, **kwargs):
     if map:
         for key in map.keys():
             if isinstance(map[key], Message):
-                map[key] = translate(map[key], context=request)
+                try:
+                    map[key] = translate(map[key], context=request)
+                except TypeError:
+                    # In zope.i18nmessageid 7.0+ the mapping is an immutable
+                    # 'mappingproxy' due to some C changes for Python 3.13
+                    # support.  Without further investigation I don't know
+                    # if this is properly fixable or even if recursiveTranslate
+                    # is no longer needed.  For now work around it:
+                    # leave this part untranslated.
+                    break
```

Comment:
@mauritsvanrees Zope messageids were always supposed to be immutable, so this change sounds like a bugfix. Probably what we should do here is construct a new mapping with all the translations, and then construct a new messageid to pass to translate.

Reply:
I had a chance to try it out for real now, with easyform.  This made me see that the recursive translation in here is not needed: it is already done in the [`translate`](https://github.com/zopefoundation/zope.i18n/blob/c2e0a66a88c71c02a391d0fd74ddedb4a71dd394/src/zope/i18n/translationdomain.py#L88-L94) function that we call from `zope.i18n`.

In fact, this has already been there since 2008. :-)
See [changelog of 3.5.0](https://github.com/zopefoundation/zope.i18n/blob/master/CHANGES.rst#350-2008-07-10).

I will close this PR and open a new one.